src/java_bytecode/Makefile

```diff
@@ -38,8 +38,18 @@ include ../common
 
 CLEANFILES = java_bytecode$(LIBEXT)
 
+clean : clean_library
+
+.PHONY: clean_library
+clean_library :
+	$(MAKE) clean -C library
```

Comment:
Please move all of the above changes below the declaration of the `all` goal so that `all` remains the default goal.